databricks/src/rf_test/train.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+from functools import partial
+
+import mlflow
+import mlflow.sklearn
+
+from cuml.metrics.accuracy import accuracy_score as cuml_acc
+from cuml.preprocessing.model_selection import train_test_split as cuml_split
+from cuml.ensemble import RandomForestClassifier as cu_RF
+
+from sklearn.model_selection import train_test_split as sk_split
+from sklearn.metrics import accuracy_score as sk_acc
+from sklearn.ensemble import RandomForestClassifier as sk_RF
+import pandas as pd
+
+from hyperopt import fmin, tpe, hp, Trials, STATUS_OK
+
+
+def load_data(fpath, compute_type):
+    data_type = 'ORC' if fpath.lower().endswith('.orc') else 'parquet'
+
+    if 'CPU' in compute_type:
+        try:
+            import pandas
+            import pyarrow
+            from pyarrow import orc
+        except Exception as error:
+            print(f'! CPU import error : {error}')
+    elif 'GPU' in compute_type:
+        try:
+            import cudf
+        except Exception as error:
+            print(f'! GPU import error : {error}')
+
+    if 'CPU' in compute_type:
+        if ('ORC' == data_type):
+            if (fpath.startswith('/') or fpath.startswith('file:')):
+                with open(fpath, mode='rb') as reader:
+                    df = pyarrow.orc.ORCFile(reader).read().to_pandas()
+            else:
+                raise NotImplemented("CPU Remote read not implemented")
+        else:
+            if (fpath.startswith('/') or fpath.startswith('file:')):
+                df = pd.read_parquet(fpath)
+            else:
+                raise NotImplemented("CPU Remote read not implemented")
+
+
+    elif ('GPU' in compute_type):
+        if ('ORC' == data_type):
+            df = cudf.read_orc(fpath)
+        else:
+            df = cudf.read_parquet(fpath)
+
+    return df
+
+
+def _train(params, fpath, mode='GPU', log_to_mlflow=None, hyperopt=False):
+    max_depth, max_features, n_estimators = params
+    max_depth, max_features, n_estimators = int(max_depth), float(max_features), int(n_estimators)
```

Comment:
how about combining these two lines?
`max_depth, max_features, n_estimators = (int(p) for p in params)`

Reply:
max_features needs to be a float here, its a scaling coefficient (0, 1.0]